### PR TITLE
Mcs/feature bounding boxes

### PR DESCRIPTION
**Pull request by freethewhat**
Added zones class to create independent zone objects. New class can draw regions and store the region of image in each object.

### CODE REVIEW

File: src/macCamera.py [hunk 1]
```diff
@@ -1,46 +1,69 @@
...
 
 while True:
     _, frame = cap.read()
     w, h, _ = frame.shape
     frame = cv2.resize(frame, (h/2,w/2))
 
     subValue = float(cv2.getTrackbarPos('Subtract', 'heatMap') / 100)
-    # frameNoiseReduction = cv2.fastNlMeansDenoisingColored(frame, None , 10, 10, 7, 21)
+
     subtract = np.full((w/2, h/2), subValue, dtype=np.uint8)
```
Comment (ed42311) [inline src/macCamera.py:39]: is this subtracting the outside edges of the frame?

File: src/macCamera.py [hunk 2]
```diff
@@ -1,46 +1,69 @@
...
 
     mask = fgbg.apply(frame)
 
     heatMap = cv2.addWeighted(heatMap, .995, mask, .005, 0);
-    heatMap = heatMap - subtract
 
+    if zone1.rectReady == True:
```
Comment (ed42311) [inline src/macCamera.py:45]: this block is nice, clear, concise and readable

File: src/zones.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+class Zone:
+    def __init__(self):
+        self.x = -1
+        self.y = -1
+        self.width = -1
+        self.height = -1
+        self.drawing = False
+        self.rectReady = False
+
+    def setSquare(self, event, x, y, flags, params):
```
Comment (ed42311) [inline src/zones.py:12]: interesting, does the user draw the square?

File: src/zones.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                self.height = y - self.y
+                print "Mouse Move: " + str(self.width) + " " + str(self.height)
+        elif event == cv2.EVENT_LBUTTONUP:
+            self.drawing = False;
+            self.rectReady = True
+            self.width = x - self.x
+            self.height = y - self.y
+            print "Mouse Up: " + str(self.width) + " " + str(self.height)
+
+    def drawSquare(self, frame):
```
Comment (ed42311) [inline src/zones.py:33]: function names are easy to understand, self!  it's just all self with python.

File: src/macCamera.py (hunk 1 above)
Comment (freethewhat) [reply src/macCamera.py:39]: this is the subtraction layer. At this time it is not in use and will need to be cleaned up.

File: src/zones.py (hunk 3 above)
Comment (freethewhat) [reply src/zones.py:12]: Yes indeed! They user will ultimately be in charge of what zones they want to monitor.

**Pull request merged**